src/qt3utils/datagenerators/princeton.py

```diff
@@ -0,0 +1,338 @@
+import os
+import logging
+import numpy as np
+from time import sleep
+from pathlib import Path
+
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger(__name__)
+
+try:
+    import clr
+
+    lf_root = Path(os.environ['LIGHTFIELD_ROOT'])
+    automation_path = lf_root / 'PrincetonInstruments.LightField.AutomationV4.dll'
+    addin_path = lf_root / 'AddInViews' / 'PrincetonInstruments.LightFieldViewV4.dll'
+    support_path = lf_root / 'PrincetonInstruments.LightFieldAddInSupportServices.dll'
+
+    addin_class = clr.AddReference(str(addin_path))
+    automation_class = clr.AddReference(str(automation_path))
+    support_class = clr.AddReference(str(support_path))
+
+    import PrincetonInstruments.LightField as lf
+
+    clr.AddReference("System.Collections")
+    clr.AddReference("System.IO")
+    from System.Collections.Generic import List
+    from System import String
+    from System.IO import FileAccess
+except KeyError as e:
+    logger.error(f"KeyError {e} during import")
+except ImportError as e:
+    logger.error(f"Unable to import packages: {e}")
+
+
+class LightfieldApplicationManager:
+    def __init__(self, visible):
+        self._addinbase = lf.AddIns.AddInBase()
+        self._automation = lf.Automation.Automation(visible, List[String]())
+        self._application = self._automation.LightFieldApplication
+        self._experiment = self._application.Experiment
+
+    @property
+    def automation(self):
+        return self._automation
+
+    @property
+    def addinbase(self):
+        return self._addinbase
+
+    @property
+    def application(self):
+        return self._application
+
+    @property
+    def experiment(self):
+        return self._experiment
+
+    def __del__(self):
+        """
+        Uses Python garbage collection method used to terminate
+        AddInProcess.exe, if this is not done, LightField will not reopen
+        properly.
+        """
+        self.automation.Dispose()
+        logger.info('Closed AddInProcess.exe')
```

Comment:
I'm suspicious that we have to call Dispose() again here. Maybe qt3scan sometimes fails to call 'close'..... to investigate, though perhaps not right now. Maybe make a note in the code that this is suspicious.